src/WebJobs.Extensions.DurableTask/Options/DurableTaskOptions.cs

```diff
@@ -191,23 +116,16 @@ internal string GetDebugString()
                 sb.Append(nameof(this.ExtendedSessionIdleTimeoutInSeconds)).Append(": ").Append(this.ExtendedSessionIdleTimeoutInSeconds).Append(", ");
             }
 
-            sb.Append(nameof(this.EventGridTopicEndpoint)).Append(": ").Append(this.EventGridTopicEndpoint).Append(", ");
-            if (!string.IsNullOrEmpty(this.EventGridTopicEndpoint))
+            if (this.Notifications != null)
             {
-                sb.Append(nameof(this.EventGridKeySettingName)).Append(": ").Append(this.EventGridKeySettingName).Append(", ");
-                sb.Append(nameof(this.EventGridPublishRetryCount)).Append(": ").Append(this.EventGridPublishRetryCount).Append(", ");
-                sb.Append(nameof(this.EventGridPublishRetryInterval)).Append(": ").Append(this.EventGridPublishRetryInterval).Append(", ");
-                sb.Append(nameof(this.EventGridPublishRetryHttpStatus)).Append(": ").Append(string.Join(", ", this.EventGridPublishRetryHttpStatus ?? new int[0])).Append(", ");
+                this.Notifications.AddToDebugString(sb);
```

Comment:
Try to maintain the nested structure by adding `{` and `}` characters where appropriate. This makes the debug string into a json representation of the host.json, which is more helpful then just listing all of the properties. See the work done for the `StorageProvider` field and the classes that compose it for examples of how to do this.

Reply:
Applied the same work done in the `StorageProvider` field, to `Notifications` and `Tracing` fields.